src/main/java/org/mifos/pheeBillPay/camel/routes/BillerFetchRouteBuilder.java

```diff
@@ -32,28 +35,55 @@ public void configure() {
             logger.debug("Bill Id: " + billId);
             BillerDetails billerDetails = getBillDetails(billId);
             if (billerDetails != null) {
-                exchange.setProperty("billerDetails", billerDetails);
-                exchange.setProperty("billerId", billerDetails.getId());
-                exchange.setProperty("billerName", billerDetails.getBiller());
-                exchange.setProperty("billerType", billerDetails.getBillerCategory());
-                exchange.setProperty("billerAccount", billerDetails.getBillerAccount());
-                exchange.setProperty("billerFetchFailed", false);
+                if(billerDetails.getId().equals(billId)) {
+                    if(billId.equals("004")){
+                        exchange.setProperty(BILL_ID,"00");
+                    }
+                    else {
+                        exchange.setProperty(BILL_ID,billId);
+                    }
+                    exchange.setProperty("billerDetails", billerDetails);
+                    exchange.setProperty("billerId", billerDetails.getId());
+                    exchange.setProperty("billerName", billerDetails.getBiller());
+                    exchange.setProperty("billerType", billerDetails.getBillerCategory());
+                    exchange.setProperty("billerAccount", billerDetails.getBillerAccount());
+                    exchange.setProperty("billerFetchFailed", false);
+                }
+                else {
+                    logger.debug("Biller details not found for bill id: " + billId);
+                    exchange.setProperty(ERROR_INFORMATION,
+                            "Unindentified Biller: Bill Id does not exist in biller table");
+                    exchange.setProperty("billerFetchFailed", true);
+                }
             } else {
                 logger.debug("Biller details not found for bill id: " + billId);
```

Comment:
Dont use + for concatenation use {} instead. This practice will make the code null safe